src/virtual_machine.rs

```diff
@@ -176,29 +186,165 @@ impl VM {
             _ => panic!("Unrecognized operation code"),
         }
     }
-}
 
-#[test]
-fn check_memory_len() {
-    let vm = VM::new();
+    fn update_flags(&mut self, value: u16) {
+        self.rcond = match value.cmp(&0) {
+            Ordering::Less => FL::NEG,
+            Ordering::Equal => FL::ZRO,
+            Ordering::Greater => FL::POS,
+        };
+    }
+
+    fn fetch_value_from_register(&self, reg_num: u16) -> u16 {
+        match reg_num {
+            0 => self.r0,
+            1 => self.r1,
+            2 => self.r2,
+            3 => self.r3,
+            4 => self.r4,
+            5 => self.r5,
+            6 => self.r6,
+            7 => self.r7,
+            8 => self.rpc,
+            10 => self.rcount,
+            _ => panic!("Invalid register"),
+        }
+    }
+
+    fn fetch_register(&mut self, reg_num: u16) -> &mut u16 {
+        match reg_num {
```

Comment:
I think the naming here is a bit confusing, there's no aparent difference between both names if you don't look at the code or the type signature.

Maybe something like `fetch_register()` `fetch_register_mut()` , so it's similar how `get` and `get_mut` work in the stdlib